syncer/dml.go

```diff
@@ -423,3 +484,94 @@ func (s *Syncer) mappingDML(schema, table string, columns []string, data [][]int
 	}
 	return rows, nil
 }
+
+// pruneGeneratedColumnDML filters columns list, data and index removing all
+// generated column. because generated column is not support setting value
+// directly in DML, we must remove generated column from DML, including column
+// list, data list and all indexes including generated columns.
+func pruneGeneratedColumnDML(columns []*column, data [][]interface{}, index map[string][]*column, schema, table string, cache *GenColCache) ([]*column, [][]interface{}, map[string][]*column, error) {
+	var (
+		cacheKey    = dbutil.TableName(schema, table)
+		cacheStatus = cache.status(cacheKey)
+	)
+
+	if cacheStatus == noGenColumn {
+		return columns, data, index, nil
+	}
+	if cacheStatus == hasGenColumn {
+		rows := make([][]interface{}, 0, len(data))
+		filters, ok := cache.isGenColumn[cacheKey]
```

Comment:
also test `cache.columns[cacheKey]` and `rows, cache.indexes[cacheKey]` here? like `ok1`, `ok2`, `ok3`.
but this maybe not a problem.